src/structures/Presence.js

```diff
@@ -31,20 +32,15 @@ const Util = require('../util/Util');
 
 /**
  * Represents a user's presence.
+ * @extends {Base}
  */
-class Presence {
+class Presence extends Base {
   /**
    * @param {Client} client The instantiating client
    * @param {APIPresence} [data={}] The data for the presence
    */
   constructor(client, data = {}) {
-    /**
-     * The client that instantiated this
-     * @name Presence#client
-     * @type {Client}
-     * @readonly
-     */
-    Object.defineProperty(this, 'client', { value: client });
+    super(client);
```

Comment:
```suggestion
    super(client);

```

Please, my heart can't take it.